File: engine/model/src/main/scala/tools/packet/inventory/ModifyInventoryPacket.scala
package tools.packet.inventory

import client.inventory.ModifyInventory
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ModifyInventoryPacket(private var _updateTick: Boolean, private var _modifications: java.util.List[ModifyInventory]) extends PacketInput {
  //noinspection MutatorLikeMethodIsParameterless
  def updateTick: Boolean = _updateTick

  def modifications: java.util.List[ModifyInventory] = _modifications

  override def opcode(): SendOpcode = SendOpcode.INVENTORY_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/inventory/ModifyInventoryPacket.java
package tools.packet.inventory; 
import java.util.List;

import client.inventory.ModifyInventory;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ModifyInventoryPacket( Boolean updateTick,  List<ModifyInventory> modifications) implements PacketInput {
  @Override
  public SendOpcode opcode() {
    return SendOpcode.INVENTORY_OPERATION;
  }
}